include/coap3/coap_pdu_internal.h

```diff
@@ -75,9 +75,9 @@
 
 #ifndef COAP_DEFAULT_MAX_PDU_RX_SIZE
 #if defined(WITH_LWIP)
-#define COAP_DEFAULT_MAX_PDU_RX_SIZE (COAP_MAX_MESSAGE_SIZE_TCP16+4UL)
+#define COAP_DEFAULT_MAX_PDU_RX_SIZE (COAP_MAX_MESSAGE_SIZE_TCP16)
```

Comment:
I think the original is still correct - it defines the maximum space for TCP (16) data (including tokens, options etc) within the PDU as well as the CoAP header being added on.

Reply:
I see. I got confused by the `assert(size <= COAP_DEFAULT_MAX_PDU_RX_SIZE);` in `pdu_init`. So, only `pdu_init` needs fixing.

Reply:
Yes, I had to think things carefully through!